ui/app/adapters/kmip/role.js

```diff
@@ -15,19 +17,47 @@ export default BaseAdapter.extend({
       return {
         id: name,
         name,
+        backend: snapshot.record.backend,
+        scope: snapshot.record.scope,
       };
     });
   },
 
+  deleteRecord(store, type, snapshot) {
+    let name = snapshot.id || snapshot.attr('name');
+    let url = this._url(
+      type.modelName,
+      {
+        backend: snapshot.record.backend,
+        scope: snapshot.record.scope,
+      },
+      name
+    );
+    return this.ajax(url, 'DELETE');
+  },
+
   serialize(snapshot) {
     // the endpoint here won't allow sending `operation_all` and `operation_none` at the same time or with
     // other values, so we manually check for them and send an abbreviated object
     let json = snapshot.serialize();
+    let keys = snapshot.record.nonOperationFields.map(decamelize);
+    let nonOperationFields = getProperties(json, keys);
+    for (let field in nonOperationFields) {
+      if (nonOperationFields[field] == null) {
+        delete nonOperationFields[field];
+      }
+    }
     if (json.operation_all) {
-      return { operation_all: true };
+      return {
+        operation_all: true,
+        ...nonOperationFields,
```

Comment:
why do we need to include the `nonOperationFields` now?

Reply:
we still need to send other things dynamically grabbed from OpenAPI (not all of them are operations)

Reply:
Yes to what @madalynrose, there were other fields added to the endpoint since I first PRd this.